src/lib/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -910,6 +911,50 @@ public function loadVersionInfo(int $contentId, ?int $versionNo = null): array
         return $queryBuilder->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    public function loadVersionNoArchivedWithin(int $contentId, int $seconds): array
+    {
+        $now = new \DateTime();
+        $interval = DateInterval::createFromDateString($seconds . ' seconds');
+        if ($interval === false) {
+            return [];
+        }
+        $now->sub($interval);
+        $queryBuilder = $this->queryBuilder->createVersionInfoFindQueryBuilder();
+        $expr = $queryBuilder->expr();
+
+        $queryBuilder
+            ->where(
+                $expr->eq(
+                    'v.contentobject_id',
+                    $queryBuilder->createNamedParameter(
+                        $contentId,
+                        ParameterType::INTEGER,
+                        ':content_id'
+                    )
+                )
+            )->andWhere(
+                $expr->eq(
+                    'v.status',
+                    $queryBuilder->createNamedParameter(
+                        VersionInfo::STATUS_ARCHIVED,
+                        ParameterType::INTEGER,
+                        ':status'
+                    )
+                )
+            )->andWhere(
+                $expr->gt(
+                    'v.modified',
+                    $queryBuilder->createNamedParameter(
+                        $now->getTimestamp(),
+                        ParameterType::INTEGER,
+                        ':modified'
+                    )
+                )
+            )->orderBy('v.modified', 'DESC');
+
+        return $queryBuilder->execute()->fetchAll(FetchMode::ASSOCIATIVE);
```

Comment:
Using flags is deprecated and removed from DBAL v3.
```suggestion
        return $queryBuilder->execute()->fetchAllAssociative();
```